File: Snapshot_GO/servent/handler/snapshot/ly_tell_handler.go
package handlersnapshot

import (
	"Snapshot/app"
	"Snapshot/app/snapshotbitcake"
	"Snapshot/servent/message"
)

type LYTellHandler struct {
	ClientMessage     message.Message
	SnapshotCollector snapshotbitcake.SnapshotCollector
}

func (lyTellHandler LYTellHandler) Run() {
	if msg, ok := lyTellHandler.ClientMessage.(*snapshotbitcake.LYTellMessage); ok {

		if msg.MessageType == "LY_TELL" {
			lyTellHandler.SnapshotCollector.AddLYSnapshotInfo(msg.OriginalSenderInfo.Id, msg.LYSnapshotResult)
		} else {
			app.TimeStampErrorPrint("Tell amount handler got: " + msg.MessageText)
		}

	} else {
		app.TimeStampErrorPrint("LYTellHandler received a Message that isn't of LYTellMessage type.")
	}
}

File: src/servent/handler/snapshot/LYTellHandler.java
package servent.handler.snapshot;

import app.AppConfig;
import app.snapshot_bitcake.SnapshotCollector;
import servent.handler.MessageHandler;
import servent.message.Message;
import servent.message.MessageType;
import servent.message.snapshot.LYTellMessage;

public class LYTellHandler implements MessageHandler {

	private Message clientMessage;
	private SnapshotCollector snapshotCollector;

	public LYTellHandler(Message clientMessage, SnapshotCollector snapshotCollector) {
		this.clientMessage = clientMessage;
		this.snapshotCollector = snapshotCollector;
	}

	@Override
	public void run() {
		if (clientMessage.getMessageType() == MessageType.LY_TELL) {
			LYTellMessage lyTellMessage = (LYTellMessage)clientMessage;
			/*snapshotCollector.addLYSnapshotInfo(
					lyTellMessage.getSnapshotNumber(),
					lyTellMessage.getOriginalSenderInfo().getId(),
					lyTellMessage.getLYSnapshotResult());
			snapshotCollector.getBitcakeManager().addIdBorderSet(lyTellMessage.getIdBorderSet());*/
		} else {
			AppConfig.timestampedErrorPrint("Tell amount handler got: " + clientMessage);
		}

	}

}
